semgrep/semgrep/rule.py

```diff
@@ -72,6 +72,13 @@ def _build_boolean_expression(
                     OPERATORS.AND, rule_raw["id"], None, rule_raw[pattern_name]
                 )
 
+        for pattern_name in pattern_names_for_operator(OPERATORS.REGEX):
+            pattern = rule_raw.get(pattern_name)
+            if pattern:
+                return BooleanRuleExpression(
```

Comment:
Is there a reason we go through BooleanRuleExpression when we don't get any of the boolean composition anyway?

Reply:
@brendongo to keep the same object representation I suppose

Reply:
We should not merge this PR without the boolean combination imo

Reply:
We could do a reverse lookup on the match to find what line it is in the buffer and have regex expose a similar output as semgrep-core.

Reply:
I updated it to include composition :+1: